fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -88,6 +97,35 @@ rocksdb::ReadOptions getReadOptions() {
 	return options;
 }
 
+ACTOR Future<Void> rocksDBMetricLogger(std::shared_ptr<rocksdb::Statistics> statistics, rocksdb::DB* db) {
+	state std::vector<std::tuple<const char*, uint32_t, uint64_t>> tickerStats = {
+		{ "StallMicros", rocksdb::STALL_MICROS, 0 },
+		{ "BytesRead", rocksdb::BYTES_READ, 0 },
+		{ "IterBytesRead", rocksdb::ITER_BYTES_READ, 0 },
+		{ "BytesWritten", rocksdb::BYTES_WRITTEN, 0 },
+	};
+	state std::vector<std::pair<const char*, std::string>> propertyStats = {
+		{ "NumCompactionsRunning", rocksdb::DB::Properties::kNumRunningCompactions },
+	};
+	loop {
+		wait(delay(SERVER_KNOBS->STORAGE_LOGGING_DELAY));
+		TraceEvent e("RocksDBMetrics");
+		for (auto& t : tickerStats) {
+			auto& [name, ticker, cum] = t;
+			uint64_t val = statistics->getTickerCount(ticker);
+			e.detail(name, val - cum);
+			cum = val;
+		}
+
+		for (auto& p : propertyStats) {
+			auto& [name, property] = p;
+			uint64_t stat = 0;
+			ASSERT(db->GetIntProperty(property, &stat));
+			e.detail(name, stat);
+		}
```

Comment:
Does it make sense to have more verbose RocksDB metrics at less frequency? We can even guard those events with a knob, and disable it by default.

Just so, we are not limited by lack of metrics.

Reply:
I added all of the logs that look relevant and don't have a performance warning.